subsys/bluetooth/host/id.c

```diff
@@ -1453,16 +1453,22 @@ uint8_t bt_read_static_addr(struct bt_hci_vs_static_addr addrs[], uint8_t size)
 		return 0;
 	}
 
-	for (i = 0; i < cnt; i++) {
-		memcpy(&addrs[i], rp->a, sizeof(struct bt_hci_vs_static_addr));
+	valid_id_cnt = ret_id_cnt;
+	for (i = 0; i < ret_id_cnt; i++) {
+		if (!BT_ADDR_IS_STATIC(&rp->a[i].bdaddr)) {
+			valid_id_cnt--;
+			continue;
+		}
```

Comment:
This is verifying a value from a vendor specific command. IMO this is better ensured by a unit test, rather than doing a dynamic verification.

Reply:
I couldn't get how UT can be used to verify that only static addresses were loaded.
IMO using UT and without dynamic verification, I can only verify that addresses returned from the controller are copied to the destination address and it totally depends on the controller response.

This function `bt_read_static_addr`, as its name implies, should only read static addresses and it should verify that.

Reply:
> it totally depends on the controller response.

That is exactly my point. This is a Zephyr vendor specific command, and thus we can assume that it's using the Zephyr controller. The verification of the addresses returned by the controller can and should be done in unit tests, or at least the dynamic verification should be done in the controller, and not the host. 

IMO when using the Zephyr vendor specific commands, the host should assume that the response is always correct, and any test or verification can and should be done at the lower layers (i.e. the controller layer).

Reply:
> > it totally depends on the controller response.
> 
> That is exactly my point. This is a Zephyr vendor specific command, and thus we can assume that it's using the Zephyr controller. The verification of the addresses returned by the controller can and should be done in unit tests, or at least the dynamic verification should be done in the controller, and not the host.
> 
> IMO when using the Zephyr vendor specific commands, the host should assume that the response is always correct, and any test or verification can and should be done at the lower layers (i.e. the controller layer).

Ok, it's clear now. Thanks for the clarification.

Reply:
> > IMO when using the Zephyr vendor specific commands, the host should assume that the response is always correct, and any test or verification can and should be done at the lower layers (i.e. the controller layer).
> 
> I'm not sure about that, the SoftDevice controller implements a bunch of those VS commands for example. Other controller vendors may want to implement some of them in the future, and we can't guarantee that their responses will be correct. IMO everything coming from HCI should be untrusted by default if we claim to support other controllers.

Hmm, yeah I've always found it odd that SoftDevice supports Zephyr VS commands. It is a fair comment Jori, but anyone using the Zephyr host should not use Zephyr VS commands with the SoftDevice controller, as we cannot provide any guarantee in those cases.

Reply:
@Thalley @jori-nordic , so what is the conclusion :) ?

Reply:
follow @Thalley 's suggestions :+1: 
don't worry about the warning, it was more a note to self than anything.

Reply:
> follow @Thalley 's suggestions +1 don't worry about the warning, it was more a note to self than anything.

So, we will trust the response from the controller and no further checks are required at the host side?

Reply:
@Thalley I dropped that change, please re-check